tutorials/single_node_tutorial/single_gpu_tutorial.ipynb

```diff
@@ -0,0 +1,3423 @@
+   "id": "d875771b",
+   "metadata": {},
+   "source": [
+    "Apply fasttext model to separate documents by their languages"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "id": "c959800c",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Reading 1 files\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Warning : `load_model` does not return WordVectorModel or SupervisedModel any more, but a `FastText` object which is very similar.\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Time taken for splitting language:147.80864667892456\n"
+     ]
+    }
+   ],
+   "source": [
+    "t0 = time.time()\n",
+    "\n",
+    "# Load dataset \n",
+    "multilingual_dataset = load_dataset(multilingual_data_path)\n",
+    "\n",
+    "#Define Language separation pipeline\n",
+    "lang_filter = FastTextLangId(os.path.join(model_path,'lid.176.bin'))\n",
+    "language_id_pipeline = ScoreFilter(lang_filter, score_field=language_field, score_type='object')\n",
+    "filtered_dataset = language_id_pipeline(multilingual_dataset)\n",
+    "\n",
+    "# The language separation pipeline will produce a result looks like ['EN',0.96873], we only want to keep the 'EN' label and drop the detailed classifier score\n",
+    "filtered_dataset.df[language_field] = filtered_dataset.df[language_field].apply(lambda score: score[1],meta = (language_field, 'object'))\n",
+    "\n",
+    "# Split the dataset to corresponding language sub-folders\n",
+    "language_stats = separate_by_metadata(filtered_dataset.df, language_separated_output_path, metadata_field=language_field).compute()\n",
+    "\n",
+    "print(f\"Time taken for splitting language:{time.time()-t0}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bd54a24a",
+   "metadata": {},
+   "source": [
+    "Load `UnicodeReformatter` to reformat any unicode appeared in the desired language dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "id": "0c09bc28",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Reading 1 files\n",
+      "Writing to disk complete for 1 partitions\n",
+      "Time taken for fixing unicode:444.5816135406494\n"
+     ]
+    }
+   ],
+   "source": [
+    "t0 = time.time()\n",
+    "\n",
+    "# Read the language specific data and fix the unicode in it\n",
+    "lang_data_path = os.path.join(language_separated_output_path, target_language)\n",
+    "lang_data = load_dataset(lang_data_path)\n",
+    "\n",
+    "cleaner = Modify(UnicodeReformatter())\n",
+    "cleaned_data = cleaner(lang_data)\n",
+    "\n",
+    "# Write the cleaned_data\n",
+    "write_to_disk(cleaned_data.df, lang_sep_cleaned_data_output_path, write_to_filename=True, output_type='jsonl')\n",
```

Comment:
Would be great if you could replace this with `cleaned_data.to_json(lang_sep_cleaned_data_output_path, write_to_filename=True)`. Let me know if you run into any issues with that.

Reply:
Fixed.